ClockDisplay.java

```diff
@@ -49,20 +59,31 @@ public void timeTick()
     {
         minutes.increment();
         if(minutes.getValue() == 0) {  // it just rolled over!
-            hours.increment();
+            if (hours.getValue()!=12){
+                hours.increment();
+                if(hours.getValue()==12){
+                    meridian = !meridian;
+                }
+            }
+            else{
+                hours.setValue(1);
+            }
```

Comment:
by monkeying around with NumberDisplay's state, you made it necessary to make this code more complex than it needs to be